x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_actions/management.js

```diff
@@ -86,15 +87,25 @@ export function actionsMenuContent(
         // the indexPattern the job was created for. An indexPattern could either have been deleted
         // since the the job was created or the current user doesn't have the required permissions to
         // access the indexPattern.
+        return item.deleting !== true && canCreateJob;
+      },
+      onClick: (item) => {
         const indexPatternNames = getIndexPatternNames();
         const jobIndicesAvailable = item.datafeedIndices.every((dfiName) => {
           return indexPatternNames.some((ipName) => ipName === dfiName);
         });
 
-        return item.deleting !== true && canCreateJob && jobIndicesAvailable;
-      },
-      onClick: (item) => {
-        cloneJob(item.id);
+        if (!jobIndicesAvailable) {
+          getToastNotifications().addDanger(
+            i18n.translate('xpack.ml.jobsList.managementActions.noSourceIndexPatternForClone', {
+              defaultMessage:
+                'Unable to clone the anomaly detection job {jobId}. No index pattern available for source index.',
```

Comment:
Related to comment above, it would be nice to get the naming of the index pattern this is not available here, as you do for the data frame analytics toast message.

Reply:
Added in 1eda34d90173330908c9fc0b5ab178a935fcbf53